src/java.desktop/share/classes/javax/swing/DebugGraphics.java

```diff
@@ -72,11 +72,22 @@ public class DebugGraphics extends Graphics {
     /**
      * Constructs a new debug graphics context that supports slowed
      * down drawing.
+     * NOTE: This constructor should not be called by
+     * Apllications, it is for Internal use only. When called directly
+     * it will create an un-usable instance.
      */
     public DebugGraphics() {
         super();
         buffer = null;
         xOffset = yOffset = 0;
+
+        //  Creates a Graphics context when Application calls the constructor
+        //  directly.
+        StackWalker walker = StackWalker.getInstance(StackWalker.Option.RETAIN_CLASS_REFERENCE);
+        if ((graphics == null) && (walker.getCallerClass() != this.getClass())) {
```

Comment:
I don't know if creating a StackWalker is expensive but I think it should be done only if graphics == null

Also the version of getInstance() being called here might throw SecurityException
https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/StackWalker.html#getInstance(java.lang.StackWalker.Option)

So you'll need to wrap it in a doPrivileged.

Reply:
Checking only if `graphics == null `will also handle the case when user sets the `graphics` to null by passing null in the constructor, so removing StackWalker and checking only for `graphics == null` will handle both the cases whether its internal class or called from other class.

Reply:
@prrace Shall I proceed by removing Stackwalker and create a graphics instance only if its `null`......?

Reply:
No, because that won't work, or more precisely will be a waste of work.
See
DebugGraphics(Graphics graphics) {
this();
this.graphics = null;
}

graphics will ALWAYS be null when you come via here which is exactly why we needed StackWalker

Reply:
Ok, then we should set graphics only If the no-args constructor is called by another program and not via by the same class.